### PR TITLE
is_contiguous and contiguous support

### PR DESCRIPTION
**Pull request by george-qi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64

Differential Revision: [D37218079](https://our.internmc.facebook.com/intern/diff/D37218079)

### CODE REVIEW

File: maskedtensor/core.py [hunk 1]
```diff
@@ -82,6 +82,8 @@ class MaskedContiguous(torch.autograd.Function):
     @staticmethod
     def forward(ctx, input):
         assert is_masked_tensor(input)
+        if input.is_contiguous():
+            return input
```
Comment (cpuhrsch) [inline maskedtensor/core.py:86]: I think this is fine, I'm not sure whether you might need to do `MaskedTensor(data, mask)`.

File: test/test_basics.py [hunk 2]
```diff
@@ -144,6 +144,43 @@ def test_to_dense_and_sparse(self):
...
+
+        self.assertEqual(contiguous_mt.is_contiguous(), True)
+        self.assertEqual(not_contiguous_mt.is_contiguous(), False)
+
+        error_msg = "MaskedTensors with sparse data do not have is_contiguous"
+        for t in [contiguous_mt_sparse, not_contiguous_mt_sparse]:
+            with self.assertRaisesRegex(ValueError, error_msg):
+                t.is_contiguous()
+            with self.assertRaisesRegex(ValueError, error_msg):
+                t.is_contiguous()
```
Comment (cpuhrsch) [inline test/test_basics.py:176]: Do you mean `t.contiguous` instead of `t.is_contiguous`?

**Review by cpuhrsch — APPROVED** [review]

File: maskedtensor/core.py [hunk 3]
```diff
@@ -306,6 +305,9 @@ def unary(cls, fn, data, mask):
 
     @classmethod
     def __torch_dispatch__(cls, func, types, args, kwargs):
+        if func is torch.ops.aten.stride:
+            return None
```
Comment (george-qi) [inline maskedtensor/core.py:309]: stride doesn't have an overloadpacket, so putting it in front

**Comment by george-qi** [comment]
@george-qi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37218079).

**Pull request closed without merging**